### ISSUE TITLE
Herb loot acting oddly

### ISSUE THREAD

**Issue by Stump876** [azerothcore/azerothcore-wotlk#12286]
### Current Behaviour

Typically I repeatedly pick an herb node to gain a skill point. I noticed some herbing nodes and not giving skill up points even if they are a color to pick (over 50 tries). Also the amount looted will vary if picked repeatedly
![image](https://user-images.githubusercontent.com/68164076/177021275-b923c3ca-f9b2-4609-bb3f-9958b7b08e16.png)
![image](https://user-images.githubusercontent.com/68164076/177021283-55c82a53-3299-4643-803d-d083501104f4.png)
![image](https://user-images.githubusercontent.com/68164076/177021295-aa704b67-81d5-4253-a95e-33a4b5d68ea3.png)
![image](https://user-images.githubusercontent.com/68164076/177021308-6a5af4da-b0c1-46f0-b48e-408dd9e0a39a.png)

### Expected Blizzlike Behaviour

Chance at skill up per node that is a color. Amount looted should not change until the herb respawns. (no source, but could easily be exploitable for max loot)

### Source

https://wowpedia.fandom.com/wiki/Profession#Increasing_professional_skill_level

### Steps to reproduce the problem

.go game 63370  or 4239 also does the same for some kingsblood and buriseweed you might need to scout around for existing nodes
.setskill 182 192 225 -not working you have to actually train herbalism from the trainer but you can .set skill so level it up
click to gather but do not loot

### Extra Notes

Noticed on Chomiecraft was able to duplicate on my local

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/0ef7f999db7284c2a7776b70c9ecf008d8551f16

### Operating system

windows 10

### Custom changes or Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-dmf-switch](https://github.com/azerothcore/mod-dmf-switch)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

**Comment by Tereneckla** [comment]
Are you seriously able to repeatedly pick on one and the same node and gain skill points?
That should not be possible

**Comment by Stump876** [comment]
Only one skill point per gathering node and only if it's a color. With the exception of skinning, you can do multplie attempts and multiple people can get a skill point from it, entire raids would do it for mining, but that was patched out in 4.0.1 One person, one point. https://github.com/chromiecraft/chromiecraft/issues/1491#issuecomment-903194332

**Comment by Tereneckla** [comment]
Where does that say a person can try again?
It only says multiple different people can gain a point

**Comment by Stump876** [comment]
>Alternatively, miners can share a node for a skill-up by not looting the vein.
On the flip side, where does it say only one attempt allowed? If multiple people can try multiple times by not looting the node, why would it not work like that for one person? Addidtionally that is how it currently works. A source would be needed to show it as a bug.

**Comment by Tereneckla** [comment]
Where does it say more than one is allowed.
Also, if the current way actually were blizzlike it would be mentioned in at least one guide out there

**Comment by neifion-00000000** [comment]
This functionality is not a bug, you should only get 1 chance per gameobject to gain skillups.

**Comment by Stump876** [comment]
Thanks for checking for me. Still the looting seems off. If I can pick and re-pick until I get 3.  Any info on that?

**Comment by neifion-00000000** [comment]
yeah I didn't read the whole issue 14 hours ago I responded to the discord thing and put my proofs here, you're correct that you shouldn't reroll loot
